lib/country_select/tag_helper.rb

```diff
@@ -59,14 +59,14 @@ def format
     end
 
     def country_options
-      country_options_for(all_country_codes, true)
+      country_options_for(all_country_codes, @options.fetch(:sort_only, ::CountrySelect::DEFAULTS[:sort_only]))
     end
 
     def all_country_codes
       codes = ISO3166::Country.codes
 
       if only_country_codes.present?
-        codes & only_country_codes
```

Comment:
For clarity, the swapping of the arrays here is what preserves the order of the `only` array. The new option is there to control if the final output is sorted by `country_options_for` or not